gateway-js/src/utilities/cleanErrorOfInaccessibleNames.ts

```diff
@@ -0,0 +1,69 @@
+import { GraphQLError, GraphQLSchema } from 'graphql';
+
+export function cleanErrorOfInaccessibleNames(
+  schema: GraphQLSchema,
+  error: GraphQLError,
+): GraphQLError {
+  const typeRegex = /"([_A-Za-z][_0-9A-Za-z]*)"/g;
+  const typeDotFieldRegex =
+    /"([_A-Za-z][_0-9A-Za-z]*)\.([_A-Za-z][_0-9A-Za-z]*)"/g;
+
+  const schemaTypes = Object.keys(schema.getTypeMap());
+  let match: RegExpExecArray | null;
+  const typeDotFieldMatches: [string, string][] = [];
+  while ((match = typeDotFieldRegex.exec(error.message)) !== null) {
```

Comment:
I'm no JavaScript regexp expert, but I think it should be possible to avoid this two stage approach by passing in a function to `replaceAll`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll#specifying_a_function_as_a_parameter

That way, you don't have to collect the matches first, then filter, and then replace them. Instead, I think you could pass a replacer function to `replaceAll` that would be called for every match. It would perform the same tests as the current code, but return a replacement string directly instead of performing a separate replace pass.

Reply:
Soooo much better!

Reply:
We had to use `replace` instead of `replaceAll` due to node 15+ requirements. This is still usable with the `g` flag on the regex - repeat tests are still passing!

Good catch @martijnwalraven.